cratetorrent/src/tracker.rs

```diff
@@ -228,6 +301,152 @@ impl Tracker {
         let resp = serde_bencode::from_bytes(&resp)?;
         Ok(resp)
     }
+
+    async fn announce_udp(&self, params: Announce) -> Result<Response> {
+        let port = thread_rng().gen_range(1025..u16::MAX);
+        let mut sock = UdpSocket::bind(SocketAddr::from(([0, 0, 0, 0], port)))
+            .await
+            .unwrap();
+
+        // All of the potential addressese of a URL
+        let mut addrs = self.url.socket_addrs(|| None).unwrap();
+        // Shuffle the list
+        addrs.shuffle(&mut thread_rng());
+
+        //TODO: Make an error for not finding an actual IPV4 address
+        let addr = *addrs.iter().find(|a| a.is_ipv4()).unwrap();
```

Comment:
I understand this is a draft, so I would just like to put a note here as a reminder to take out the unwrap once the PR nears completion.

Reply:
Oh yea, of course. I definitely won't resolve this until I actually fix it.